.github/workflows/main.yml

```diff
@@ -70,12 +73,40 @@ jobs:
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build image for tests
+        uses: docker/build-push-action@v4.0.0
+        with:
+          load: true
+          build-args: |
+            no-cache=true
+            JDIST=${{ matrix.jdist }}
+            DISTRIB_NAME=${{ matrix.base_image.flavor }}
+            DISTRIB_MAJOR=${{ matrix.base_image.major }}
+            JAVA_MAJOR=${{ matrix.java_major }}
+            TOMCAT_MAJOR=${{ matrix.tomcat_major }}
+          tags: local/${{ env.IMAGE_REPOSITORY }}:ci
+          target: TOMCAT_BASE_IMAGE
+      - name: Test Built Image
+        run: |
+          echo -n "Checking for Tomcat config: "
+          docker run local/${{ env.IMAGE_REPOSITORY }}:ci ./bin/catalina.sh configtest 2>&1 \
+          | tail -1 | grep '^INFO: Server initialization in '
+          echo -n "Checking for tcNative libs: "
+          docker run local/${{ env.IMAGE_REPOSITORY }}:ci ./bin/catalina.sh configtest 2>&1 \
```

Comment:
better capture the output of the previous command instead of running container two times for the same command?